plugins/outputs/http/http.go

```diff
@@ -168,6 +168,18 @@ func (h *HTTP) writeMetric(reqBody []byte) error {
 		req.SetBasicAuth(h.Username, h.Password)
 	}
 
+	// google api auth
+	if h.HTTPClientConfig.OAuth2Config.CredentialsFile != "" {
+		if !h.HTTPClientConfig.OAuth2Config.AccessToken.Valid() {
+			err := h.HTTPClientConfig.OAuth2Config.GetAccessToken(context.Background(), h.URL)
+			if err != nil {
+				return err
+			}
+		}
+
+		h.HTTPClientConfig.OAuth2Config.AccessToken.SetAuthHeader(req)
+	}
```

Comment:
Maybe just one more suggestion... If you move the token validity check into `GetAccessToken` and make that function return the token, you can mimic the other code around this function, i.e.
```suggestion
	if h.HTTPClientConfig.OAuth2Config.CredentialsFile != "" {
		token, err := h.HTTPClientConfig.OAuth2Config.GetAccessToken(context.Background(), h.URL)
		if err != nil {
			return err
		}
		token.SetAuthHeader(req)
	}
```

It would hide the whole refresh logic in `h.HTTPClientConfig.OAuth2Config` and you can also "unexport" `AccessToken` avoiding direct manipulation.

What do you think? I would not hold up the whole thing just for that, but it looks cleaner to me...

Reply:
Once again, great suggestion!